x/circuit/keeper/msg_server.go

```diff
@@ -70,40 +75,52 @@ func (srv msgServer) AuthorizeCircuitBreaker(goCtx context.Context, msg *types.M
 	}, nil
 }
 
-func (srv msgServer) TripCircuitBreaker(goCtx context.Context, msg *types.MsgTripCircuitBreaker) (*types.MsgTripCircuitBreakerResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
-
+func (srv msgServer) TripCircuitBreaker(ctx context.Context, msg *types.MsgTripCircuitBreaker) (*types.MsgTripCircuitBreakerResponse, error) {
 	address, err := srv.addressCodec.StringToBytes(msg.Authority)
 	if err != nil {
 		return nil, err
 	}
 
 	// Check that the account has the permissions
-	perms, err := srv.GetPermissions(ctx, address)
-	if err != nil {
-		return nil, fmt.Errorf("user permission does not exist %w", err)
+	perms, err := srv.Permissions.Get(ctx, address)
+	if err != nil && !errorsmod.IsOf(err, collections.ErrNotFound) {
+		return nil, err
 	}
 
-	store := ctx.KVStore(srv.storekey)
-
 	switch {
 	case perms.Level == types.Permissions_LEVEL_SUPER_ADMIN || perms.Level == types.Permissions_LEVEL_ALL_MSGS || bytes.Equal(address, srv.GetAuthority()):
 		for _, msgTypeURL := range msg.MsgTypeUrls {
 			// check if the message is in the list of allowed messages
-			if !srv.IsAllowed(ctx, msgTypeURL) {
+			isAllowed, err := srv.IsAllowed(ctx, msgTypeURL)
+			if err != nil {
+				return nil, err
+			}
+
+			if !isAllowed {
 				return nil, fmt.Errorf("message %s is already disabled", msgTypeURL)
 			}
-			store.Set(types.CreateDisableMsgPrefix(msgTypeURL), []byte{0x01})
+
+			if err = srv.DisableList.Set(ctx, msgTypeURL); err != nil {
```

Comment:
We should use the keeper method doing that. Or remove that keeper helper.

Reply:
I'll remove the helper 👌